platform/FileBase.cpp

```diff
@@ -52,45 +55,53 @@ FileBase::~FileBase()
             p->_next = _next;
         }
     }
-    _mutex->unlock();
+
+    if (_default == this) {
+        _default == NULL;
+    }
 
     if (getPathType() == FilePathType) {
-        extern void remove_filehandle(FileHandle * file);
-        remove_filehandle(static_cast<FileHandle *>(static_cast<FileLike *>(this)));
+        extern void remove_filehandle(FileHandle *file);
+        remove_filehandle(static_cast<FileLike *>(this));
     }
 }
 
+void FileBase::set_as_default()
+{
+    ScopedLock<PlatformMutex> lock(*_mutex);
+    _default = this;
+}
+
 FileBase *FileBase::lookup(const char *name, unsigned int len)
 {
-    _mutex->lock();
+    ScopedLock<PlatformMutex> lock(*_mutex);
     FileBase *p = _head;
     while (p != NULL) {
         /* Check that p->_name matches name and is the correct length */
-        if (p->_name != NULL && std::strncmp(p->_name, name, len) == 0 && std::strlen(p->_name) == len) {
-            _mutex->unlock();
+        if (p->_name != NULL && std::memcmp(p->_name, name, std::strlen(p->_name)) == 0) {
             return p;
         }
         p = p->_next;
     }
-    _mutex->unlock();
+    if (len == (sizeof "default") - 1 && std::memcmp("default", name, len) == 0) {
+        return _default;
+    }
     return NULL;
 }
 
 FileBase *FileBase::get(int n)
 {
-    _mutex->lock();
+    ScopedLock<PlatformMutex> lock(*_mutex);
```

Comment:
This is the wrong PR to change these locking calls.

I'm not opposed to this change, but every change invites bugs.

Reply:
You're not wrong - I certainly wouldn't do this in a patch/fix. Probably shouldn't be doing it in rc2 either.

I can undo this, but as the current test set has passed, and we might be accepting this version otherwise, is it worth restarting CI just to take it out? @0xc0170 ?

I'll prepare a removal anyway in case I do another spin, but won't push yet.

Reply:
I agree not to block it now

Reply:
Actually another CI round is required anyway for part 2 - #7977, so we could still change it.